File: backend/src/main/java/com/yourowncrm/service/AppointmentService.java
package com.yourowncrm.service;

import com.yourowncrm.dto.request.AppointmentRequest;
import com.yourowncrm.dto.response.AppointmentResponse;
import com.yourowncrm.model.Appointment;
import com.yourowncrm.dto.response.AvailabilityConflict;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface AppointmentService {
    List<AppointmentResponse> getDailySchedule(UUID tenantId, LocalDate date, Long locationId);
    AppointmentResponse       getById(UUID tenantId, Long id);
    AppointmentResponse       create(UUID tenantId, Long userId, AppointmentRequest req);
    AppointmentResponse       update(UUID tenantId, Long id, Long userId, AppointmentRequest req);
    void                      cancel(UUID tenantId, Long id, Long userId);
    AvailabilityConflict      checkAvailability(UUID tenantId, AppointmentRequest req);

    AppointmentResponse toResponsePublic(Appointment a);
}

File: frontend/src/app/core/services/appointment.service.ts
import { Injectable } from "@angular/core";
import { HttpClient, HttpParams } from "@angular/common/http";
import { Observable } from "rxjs";
import { environment } from "../../../environments/environment";
import { Appointment, AppointmentRequest, AvailabilityConflict } from "../../shared/models/appointment.model";

@Injectable({ providedIn: "root" })
export class AppointmentService {
  private base = `${environment.apiUrl}/appointments`;

  constructor(private http: HttpClient) {}

  getDailySchedule(date: string, locationId?: number): Observable<Appointment[]> {
    let p = new HttpParams().set("date", date);
    if (locationId) p = p.set("locationId", locationId);
    return this.http.get<Appointment[]>(`${this.base}/daily`, { params: p });
  }

  checkAvailability(req: AppointmentRequest): Observable<AvailabilityConflict> {
    return this.http.post<AvailabilityConflict>(`${this.base}/check-availability`, req);
  }

  create(req: AppointmentRequest): Observable<Appointment> {
    return this.http.post<Appointment>(this.base, req);
  }

  update(id: number, req: AppointmentRequest): Observable<Appointment> {
    return this.http.put<Appointment>(`${this.base}/${id}`, req);
  }

  cancel(id: number): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
